scalableminds/webknossos
# Allow to create an organization via UI / Onboarding Flow

### URL of deployed dev instance (used for testing):
- https://onboarding.webknossos.xyz

### Steps to test:
- for feedback on text, use https://onboarding.webknossos.xyz/onboarding

- for functionality test, check out locally and follow this procedure:
- ensure the following properties in application.conf:
```
  insertInitialData = false
  allowOrganzationCreation = false
```
- drop the db
- start wk
- you should be redirected to the onboarding flow
- after choosing a org name and filling out the registration form, you should be logged in when the view is switched to the upload view
- opening the onboarding view and creating a new admin account should **not** work (since allowOrganzationCreation = false)
- set allowOrganzationCreation = true, and try to create a second organization --> should work
- set allowOrganzationCreation = false, set the superUser flag in the db and try to create a second organization --> should work

@fm3 Can you review the backend part, please?

### Issues:
- fixes #2851

------
- [X] Ready for first feedback: Please give feedback on the text + links of the onboarding flow. Complete code clean up will be done afterwards.
- [ ] Ready for final review
